addons/sale_stock/models/res_users.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class Users(models.Model):
+    _inherit = ['res.users']
+
+    warehouse_id_visibility = fields.Boolean(compute='_compute_warehouse_id_visibility')
+    warehouse_id_editability = fields.Boolean(compute='_compute_warehouse_id_editability')
+    property_warehouse_id = fields.Many2one('stock.warehouse', string='Default Warehouse', company_dependent=True, check_company=True)
+
+    def _compute_warehouse_id_visibility(self):
+        self.warehouse_id_visibility = False
+        if self.user_has_groups('stock.group_stock_multi_locations'):
+            if self.id == self.env.user.id and self.user_has_groups('stock.group_stock_user,sales_team.group_sale_salesman,base.group_erp_manager'):
```

Comment:
```suggestion
            if self == self.env.user and has_low_access_rights_groups:
```
And as user_has_groups is quite heavy, before the for loop, compute
```python3
has_low_access_rights_groups 
 = self.user_has_groups('stock.group_stock_user,sales_team.group_sale_salesman,base.group_erp_manager')
```